src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,198 @@
+        init = atoms
+        initial_atoms: FrozenSet[GroundAtom] = frozenset(atoms)
+        objects = set(state)
+        # Dictionary used to create policy for search function.
+        dictionary = self._get_dictionary(initial_atoms, atoms, goal, state)
+        ground_nsrts = [
+            ground_nsrt for nsrt in self._get_current_nsrts()
+            for ground_nsrt in utils.all_ground_nsrts(nsrt, objects)
+        ]
+
+        
+        action_seq = self._run_policy_guided_planning(dictionary,goal,ground_nsrts,
+        init,state,initial_atoms)
+        return action_seq
+
+    def _get_llm_based_state_action_predictions(
+            self, atoms: Set[GroundAtom], goal: Set[GroundAtom],
+            state: State) -> List[Tuple[List[FrozenSet[GroundAtom]], 
+            List[Optional[Dict[FrozenSet[GroundAtom], _GroundNSRT]]]]]:
+        trajectories = []
+        initial_atoms = atoms
+
+        new_prompt = self._create_prompt(atoms, goal, [])
+        prompt = self._prompt_prefix + new_prompt
+        llm_predictions = self._llm.sample_completions(
+            prompt=prompt,
+            temperature=CFG.open_loop_llm_temperature,
+            seed=CFG.seed,
+            num_completions=CFG.open_loop_llm_num_completions)
+
+        for llm_prediction in llm_predictions:
+            objects = set(state)
+            option_plan = self._llm_prediction_to_option_plan(
+                llm_prediction, objects)
+            if len(option_plan) == 0:
+                continue
+            states, actions = self._get_nsrt_ground_plan(
+                initial_atoms, option_plan)
+
+            trajectories.append((states, actions))
+        return trajectories
+
+    def _get_nsrt_ground_plan(
+        self, initial_atoms: Set[GroundAtom],
+        option_plan: List[Tuple[ParameterizedOption, Sequence]]
+    ) -> Tuple[List, List]:
+
+        states = []
+        actions = []
+        states.append(initial_atoms)
+        nsrts = self._get_current_nsrts()
+        # We assume that operators and NSRTs are the same
+        # for our use case, so this function would break down
+        # for environments not written in PDDL.
+        ground_nsrt_plan = []
+        for (option, objects) in option_plan:
+            for n in nsrts:
+                if option.name == n.option.name:
+                    ground_nsrt_plan.append(n.ground(objects))
+                    break
+        atoms = initial_atoms
+        for nsrt in ground_nsrt_plan:
+            atoms = utils.apply_operator(nsrt, atoms)
+            actions.append(nsrt)
+            states.append(atoms)
+        return states, actions
+
+    def _get_dictionary(
+        self, initial_atoms: Set[GroundAtom], atoms: Set[GroundAtom],
+        goal: Set[GroundAtom],state:State
+    ) -> Optional[dict]:
+        
+        trajectories = self._get_llm_based_state_action_predictions(
+            atoms, goal, state)
+        final_states = []
+        final_actions = []
```

Comment:
okay so the logic in this function is looking pretty complicated and I'm hoping we can simplify it. could we do something like the following? (not tested)
```python
abstract_policy_dict = {}
for (states, actions) in trajectories:
    assert len(states) == len(actions) + 1
    for (s, a) in zip(states, actions):
        abstract_policy_dict[s] = a
return abstract_policy_dict
```
does that miss any important cases that are handled by the existing code?